jax/_src/config.py

```diff
@@ -1501,6 +1501,11 @@ def _update_disable_jit_thread_local(val):
     upgrade=True,
     help='Enable eager-mode pmap when jax_disable_jit is activated.')
 
+no_tracing = bool_state(
+    name='jax_no_tracing',
+    default=False,
+    help='Disallow tracing for JIT compilation.')
```

Comment:
You might wanna add (after it's traced for the first time?)

Reply:
/I think this should (and does?) trigger even if it's the first time